### DISCUSSION TITLE
Warnings in the CodeQL Analysis workflow

### DISCUSSION THREAD

**Discussion by sinsukehlab** [skills/secure-code-game#87] [Ideas]
Hi, I would like to update the versions of actions used in the GitHub Actions workflows to resolve the warnings related to Node.js 16 deprecation. I have opened issue #86.

**Comment by sinsukehlab** [comment]
@jkcso @hectorsector I would also like to add dependabot.yml to this repo like other skills template repo. I'm wondering if there will be any problems with it.

**Reply by jkcso** [reply]
Go for it!

**Reply by sinsukehlab** [reply]
Thanks. I have opened PR #89.

**Reply by sinsukehlab** [reply]
Thank you for merging my PR.
@dependabot will bump secure-code-game-action in a PR. Feel free to merge it.
The action still uses Node.js 16. Is this deliberate?

**Reply by jkcso** [reply]
I wouldn't say it's deliberate. Do you refer to a specific level of the game?

**Reply by sinsukehlab** [reply]
Yes, I refer to Season 2 Level 1.
The action [dduzgun-security/secure-code-game-action](/dduzgun-security/secure-code-game-action) used in the workflow jarvis-code.yml still uses Node.js 16.
https://github.com/dduzgun-security/secure-code-game-action/blob/b30b00ac91e4cbd0a8636ccbef1f38f4ffc376ea/action.yml#L16-L18
```yml
runs:
  using: 'node16'
  main: 'index.js'
```

**Reply by jkcso** [reply]
Let's ask the author of the level, the legend himself! @dduzgun-security

**Reply by dduzgun-security** [reply]
Thanks for the issue and [PR](https://github.com/dduzgun-security/secure-code-game-action/pull/6) to fix it 💪 we may need to change the GitHub Action [here](https://github.com/skills/secure-code-game/blob/main/.github/workflows/jarvis-code.yml#L30) to point out to the new commit SHA `1c9ed9f1e57d7b8c4e9bfa8013fd54e322214eb4` of the [updated available action `v2.0` using node20](https://github.com/dduzgun-security/secure-code-game-action/releases/tag/v2.0).

https://github.com/skills/secure-code-game/pull/92

**Reply by sinsukehlab** [reply]
I think dduzgun-security/secure-code-game-action@1c9ed9f (dduzgun-security/secure-code-game-action#6) should be tagged with v1.1 rather than v2.0 because this update is not a major update.
See: https://github.com/sinsukehlab/secure-code-game-action/tags
